### ISSUE TITLE
No Auth option

### ISSUE THREAD

**Issue by gwharton** [swissup/module-email#11]
Is it possible to get support for no authentication.

Google's mail relay has the option of setting up authentication by IP address of server, so no need to store smtp account password in Magento.

Would be great to see this added to the Auth list.

**Comment by gwharton** [comment]
Yes, thats exactly what I mean.

Users with GSuite account are allowed to use smtp-relay.gmail.com and have the ability to setup the authentication mechanisms on that service. One of them is authenticate by IP. You add the list of IP addresses in your GSuite admin pages that are allowed to relay email via smtp-relay.gmail.com and it effectively becomes an open relay with no authentication required, but only on connections from those IP's.

It means I can change my GSuite password at will, and I don't have to go round all my servers and update their config with new account details. It also means my password isnt stored in the config of the server.

I just need this module to support "No authentication", just a plain old SMTP connection.

I'll gladly prepare a PR for the necessary changes if that would help.

**Comment by 0m3r** [comment]
PRs Welcome

Unfortunately I don't have prepared SMTP relay setting.

p.s.
Did I move in the right direction?
~~~diff
diff --git a/Mail/Transport/Smtp.php b/Mail/Transport/Smtp.php
index 8a31a81..d434546 100644
--- a/Mail/Transport/Smtp.php
+++ b/Mail/Transport/Smtp.php
@@ -42,14 +42,22 @@ class Smtp extends \Zend\Mail\Transport\Smtp implements TransportInterface
             if ($ssl) {
                 $connectionConfig['ssl'] = $ssl;
             }
-            $options = new SmtpOptions(
-                [
-                    'host' => $config['host'],
-                    'port' => $config['port'],
-                    'connection_class' => $config['auth'],//'login',
-                    'connection_config' => $connectionConfig
-                ]
-            );
+            $options = [
+                'host' => $config['host'],
+                'port' => $config['port']
+            ];
+
+            if ($config['auth'] !== '') {
+                $options['connection_class'] = $config['auth'];
+                $options['connection_config'] = connectionConfig;
+            }
+
+            $options = new SmtpOptions($options);
+
+            if ($config['auth'] !== '') {
+                $ip = $config['user'];
+                $options->setName($ip);
+            }

~~~

**Comment by gwharton** [comment]
See PR #12

**Comment by gwharton** [comment]
@0m3r I have created a pull request, pretty much inline with your suggestion above, with some minor changes.

I have tested it with a server requiring SMTP auth, and a server not requiring SMTP auth. I can confirm it works with smtp-relay.gmail.com if you have configured the google service to not require authentication (authenticate by IP address instead of credentials) and you set this module to Authentication type "None".

Existing installations should not be affected by this change and should continue as normal. It should only be when adding a new service or editing an existing service that the user should now see the option of "None" in authentication type.

Hope this is ok.
